cli/src/action/circuit/builder.rs

```diff
@@ -377,20 +392,52 @@ fn is_match(service_id_match: &str, service_id: &str) -> bool {
     })
 }
 
-fn make_splinter_node(node_id: &str, endpoints: &[String]) -> Result<SplinterNode, CliError> {
-    let node = SplinterNodeBuilder::new()
+fn make_splinter_node(
+    node_id: &str,
+    endpoints: &[String],
+    #[cfg(feature = "challenge-authorization")] public_key: Option<&String>,
+) -> Result<SplinterNode, CliError> {
+    #[allow(unused_mut)]
+    let mut node_builder = SplinterNodeBuilder::new()
         .with_node_id(&node_id)
-        .with_endpoints(endpoints)
-        .build()
-        .map_err(|err| {
-            CliError::ActionError(format!(
-                "Failed to build node: {}",
-                msg_from_builder_error(err)
-            ))
-        })?;
+        .with_endpoints(endpoints);
+
+    #[cfg(feature = "challenge-authorization")]
+    {
+        if let Some(public_key) = public_key {
+            node_builder = node_builder.with_public_key(&parse_hex(&public_key)?)
+        }
+    }
```

Comment:
If public keys are required, should this error if there isn't a public key provided? I noticed while testing that I could specify the `challenge` auth_type without adding the `--node-public-key` option for either node. That proposed circuit would have the challenge auth type. (And I'm able to create a circuit from that)

Reply:
Nevermind, realized there's another PR for this concern